bundle/bundle_test.go

```diff
@@ -680,3 +681,63 @@ func TestImageDeepCopy(t *testing.T) {
 	assert.Equal(t, map[string]string{"origLabel": "newLabelValue"}, newImg.Labels)
 	assert.Equal(t, "123abcd", newImg.Digest)
 }
+
+func ExampleBaseImage_DigestedRef() {
+	image := BaseImage{
+		Image:  "foo/bar:latest",
+		Digest: "sha256:7ea254518cab82f53938523eab0dc6601a3dd1edf700f1f6235e66e27cbc7986",
+	}
+
+	imageRef, ok := image.DigestedRef()
+	if !ok {
+		fmt.Println("no digest")
+	}
+
+	fmt.Println(imageRef)
+	// Output: foo/bar:latest@sha256:7ea254518cab82f53938523eab0dc6601a3dd1edf700f1f6235e66e27cbc7986
+}
+
+func TestBaseImage_DigestedRef(t *testing.T) {
+	cases := map[string]struct {
+		expected  string
+		hasDigest bool
+		image     BaseImage
+	}{
+		"no tag": {
+			expected: "foo",
+			image: BaseImage{
+				Image: "foo",
+			},
+		},
+		"tag only": {
+			expected: "foo:bar",
+			image: BaseImage{
+				Image: "foo:bar",
+			},
+		},
+		"digest only": {
+			expected:  "foo@sha256:7ea254518",
```

Comment:
I think it would be better to use a full SHA 256 hash to avoid any confusion. Maybe just pad it with zeros?